astroid/bases.py

```diff
@@ -543,9 +543,25 @@ class Generator(BaseInstance):
 
     special_attributes = util.lazy_descriptor(objectmodel.GeneratorModel)
 
-    def __init__(self, parent=None):
+    def __init__(self, parent=None, generator_initial_context=None):
         super().__init__()
         self.parent = parent
+        self._call_context = contextmod.copy_context(generator_initial_context)
+
+    @decorators.cached
+    def infer_yield_types(self):
+        # pylint: disable=import-outside-toplevel; circular import
+
+        from astroid import node_classes
```

Comment:
The fact that it's adding a circular import is problematic for me, we're trying to get rid of them, and this look like a real one here. What could we do to avoid that ?

Reply:
I'm not sure, I try to think of an improvement, maybe a function object may have a method to return 
Also, I will mention that a very similar flow exists on `_infer_type_new_call` in the same module.

Reply:
I've send a different implementation that does not requires a circular import.

Thanks!

Reply:
Well, I looked into it, seems like the problem is probably due to the file structure with too much content not well organized. We'll have to reorganize later for the other circular import anyway.

Reply:
Nevermind, thank you for the fix, very clean ! :)